### ISSUE TITLE
Is PMIX_NODE_SIZE to be kept up-to-date?

### ISSUE THREAD

**Issue by krempel-pt** [pmix/pmix-standard#401]
`PMIX_NODE_SIZE` is described as "Number of processes across all jobs that are executing upon the node."

First question is if this means "client processes running on the node", or "clients registered to run on the node"?

In the first case, that would mean, that the host environment shall pass the number of processes that are already running in other jobs but not count the processes that will run in the namespace's job, since at namespace registration time they cannot be already started, right? 

Furthermore, it is unclear, whether `PMIX_NODE_SIZE` is an information one can rely on to be up-to-date at any time?

And if that is the case it is unclear who's duty it is to take care about that. Effectively it has to be updated for all effected nodes on every spawn or termination of client processes.

If the answer to the first question above is "clients registered", then the spawn case is most probably covered by the host environment by passing the current values with the namespace registration. But is this automatically to be updated by the server library on namespace deregistration or shall the host environment do that e.g. by calling `PMIx_server_register_resources()` with `PMIX_NODE_INFO` containing the updated `PMIX_NODE_SIZE` for each effected node? 

If the answer to the first question is "processes currently running", then client connections could be counted by the server library. But has this to be done there? 

### References
6.1.5
16.2.3

**Comment by rhc54** [comment]
I think `PMIX_NODE_INFO` got used in a few places where you meant `PMIX_NODE_SIZE`. Setting that aside, the `PMIX_NODE_SIZE` is as defined - the number of processes (whether they are using PMIx or not) started by the host environment on the node. It does not include system processes, for example, but should include all user-level applications.

The only one who can take care of it is the host - PMIx has no idea what the host has started. We only know what was registered. In an ideal world, the host would register everything it starts, but there is no rule mandating that they do so, nor any way that PMIx could enforce such a rule. Likewise, the PMIx server cannot count proc connections as not every proc started by the host will call `PMIx_Init`. So we rely on the host to tell us this value at time of registration.

While a PMIx library could update its own local `PMIX_NODE_SIZE` as clients terminate or are registered, we have no way of doing so for a remote node. Thus, any updates have to come from the host. That said, I'm unaware of anyone doing that today as it isn't a trivial thing to do (especially in a highly dynamic environment). We therefore generally tell people that the number is a "snapshot in time" taken at the moment of registration - the actual number of procs running on a given node is a continually fluctuating value. If you want an update of the snapshot, you probably should use `PMIx_Query_info` to request it - again with the understanding that the value may be out-of-date by the time you get it.

The purpose of the value is to provide an app with some idea of the load on the current node so it can (if it chooses) do some resource management. Not perfect, but perhaps better than nothing.

**Comment by krempel-pt** [comment]
Ok, I think I got that now, thanks for the detailed explanation.

I corrected the typos in the first post, thanks.

I would suggest to add one sentence to the `PMIX_NODE_SIZE` description clarifying that all user level processes managed by the host environment are counted, independent of whether they are using PMIx or not. At least for me the "across all jobs" turned into something like "across all PMIx jobs" inside of my head giving me the wrong idea.
